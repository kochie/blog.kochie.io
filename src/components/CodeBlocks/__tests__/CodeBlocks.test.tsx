import React, { Suspense } from 'react'
import { ReactTestRenderer, act, create } from 'react-test-renderer'

describe('CODEBLOCKS COMPONENT', () => {
  test('renders correctly', async () => {
    const CodeBlock = (await import('@/components/CodeBlocks/codeblock'))
      .default

    let tree: ReactTestRenderer

    await act(async () => {
      tree = create(
        <Suspense>
          <CodeBlock className={'language-typescript{10}'} />
        </Suspense>
      )
    })

    // @ts-expect-error tree will be assigned
    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('renders highlight', async () => {
    const CodeBlock = (await import('@/components/CodeBlocks/codeblock'))
      .default

    let tree: ReactTestRenderer

    await act(async () => {
      tree = create(
        <Suspense>
          <CodeBlock className={'language-typescript{10}'} />
        </Suspense>
      )
    })

    // @ts-expect-error tree will be assigned
    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('renders highlight range', async () => {
    const CodeBlock = (await import('@/components/CodeBlocks/codeblock'))
      .default

    let tree: ReactTestRenderer

    await act(async () => {
      tree = create(
        <Suspense>
          <CodeBlock className={'language-typescript{10-15}'} />
        </Suspense>
      )
    })

    // @ts-expect-error tree will be assigned
    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('renders line numbers', async () => {
    const CodeBlock = (await import('@/components/CodeBlocks/codeblock'))
      .default

    let tree: ReactTestRenderer

    await act(async () => {
      tree = create(
        <Suspense>
          <CodeBlock className={'language-typescript{}[LineNumbers]'} />
        </Suspense>
      )
    })

    // @ts-expect-error tree will be assigned
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
