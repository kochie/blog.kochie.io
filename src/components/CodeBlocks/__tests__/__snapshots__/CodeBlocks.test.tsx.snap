// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CODEBLOCKS COMPONENT renders correctly 1`] = `
<div
  className="my-5 relative group"
>
  <pre
    className="prism-code language-typescript code relative rounded-lg"
    style={
      {
        "backgroundColor": "#011627",
        "color": "#d6deeb",
      }
    }
  >
    <div
      aria-label="Copy to clipboard"
      className="absolute top-3 right-3 group-hover:opacity-100 opacity-0 transform-gpu transition duration-200 group-hover:block"
      title="Copy to clipboard"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-copy fa-fw fa-xl cursor-pointer p-2 bg-slate-500 hover:bg-slate-600 border-2 border-slate-500 hover:border-slate-800 duration-200 rounded-lg active:bg-slate-50"
        data-icon="copy"
        data-prefix="fad"
        focusable="false"
        onClick={[Function]}
        role="img"
        style={{}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          className="fa-duotone-group"
          style={{}}
        >
          <path
            className="fa-secondary"
            d="M192 128H48c-26.5 0-48 21.5-48 48V464c0 26.5 21.5 48 48 48H272c26.5 0 48-21.5 48-48V416H256v32H64V192H192V128z"
            fill="currentColor"
            style={{}}
          />
          <path
            className="fa-primary"
            d="M224 48c0-26.5 21.5-48 48-48H396.1c12.7 0 24.9 5.1 33.9 14.1l67.9 67.9c9 9 14.1 21.2 14.1 33.9V336c0 26.5-21.5 48-48 48H272c-26.5 0-48-21.5-48-48V48z"
            fill="currentColor"
            style={{}}
          />
        </g>
      </svg>
    </div>
    <div
      className="token-line"
      style={
        {
          "backgroundColor": undefined,
          "color": "#d6deeb",
        }
      }
    >
      <span>
        <span
          className="token plain"
          style={
            {
              "display": "inline-block",
            }
          }
        >
          

        </span>
      </span>
    </div>
  </pre>
</div>
`;

exports[`CODEBLOCKS COMPONENT renders highlight 1`] = `
<div
  className="my-5 relative group"
>
  <pre
    className="prism-code language-typescript code relative rounded-lg"
    style={
      {
        "backgroundColor": "#011627",
        "color": "#d6deeb",
      }
    }
  >
    <div
      aria-label="Copy to clipboard"
      className="absolute top-3 right-3 group-hover:opacity-100 opacity-0 transform-gpu transition duration-200 group-hover:block"
      title="Copy to clipboard"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-copy fa-fw fa-xl cursor-pointer p-2 bg-slate-500 hover:bg-slate-600 border-2 border-slate-500 hover:border-slate-800 duration-200 rounded-lg active:bg-slate-50"
        data-icon="copy"
        data-prefix="fad"
        focusable="false"
        onClick={[Function]}
        role="img"
        style={{}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          className="fa-duotone-group"
          style={{}}
        >
          <path
            className="fa-secondary"
            d="M192 128H48c-26.5 0-48 21.5-48 48V464c0 26.5 21.5 48 48 48H272c26.5 0 48-21.5 48-48V416H256v32H64V192H192V128z"
            fill="currentColor"
            style={{}}
          />
          <path
            className="fa-primary"
            d="M224 48c0-26.5 21.5-48 48-48H396.1c12.7 0 24.9 5.1 33.9 14.1l67.9 67.9c9 9 14.1 21.2 14.1 33.9V336c0 26.5-21.5 48-48 48H272c-26.5 0-48-21.5-48-48V48z"
            fill="currentColor"
            style={{}}
          />
        </g>
      </svg>
    </div>
    <div
      className="token-line"
      style={
        {
          "backgroundColor": undefined,
          "color": "#d6deeb",
        }
      }
    >
      <span>
        <span
          className="token plain"
          style={
            {
              "display": "inline-block",
            }
          }
        >
          

        </span>
      </span>
    </div>
  </pre>
</div>
`;

exports[`CODEBLOCKS COMPONENT renders highlight range 1`] = `
<div
  className="my-5 relative group"
>
  <pre
    className="prism-code language-typescript code relative rounded-lg"
    style={
      {
        "backgroundColor": "#011627",
        "color": "#d6deeb",
      }
    }
  >
    <div
      aria-label="Copy to clipboard"
      className="absolute top-3 right-3 group-hover:opacity-100 opacity-0 transform-gpu transition duration-200 group-hover:block"
      title="Copy to clipboard"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-copy fa-fw fa-xl cursor-pointer p-2 bg-slate-500 hover:bg-slate-600 border-2 border-slate-500 hover:border-slate-800 duration-200 rounded-lg active:bg-slate-50"
        data-icon="copy"
        data-prefix="fad"
        focusable="false"
        onClick={[Function]}
        role="img"
        style={{}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          className="fa-duotone-group"
          style={{}}
        >
          <path
            className="fa-secondary"
            d="M192 128H48c-26.5 0-48 21.5-48 48V464c0 26.5 21.5 48 48 48H272c26.5 0 48-21.5 48-48V416H256v32H64V192H192V128z"
            fill="currentColor"
            style={{}}
          />
          <path
            className="fa-primary"
            d="M224 48c0-26.5 21.5-48 48-48H396.1c12.7 0 24.9 5.1 33.9 14.1l67.9 67.9c9 9 14.1 21.2 14.1 33.9V336c0 26.5-21.5 48-48 48H272c-26.5 0-48-21.5-48-48V48z"
            fill="currentColor"
            style={{}}
          />
        </g>
      </svg>
    </div>
    <div
      className="token-line"
      style={
        {
          "backgroundColor": undefined,
          "color": "#d6deeb",
        }
      }
    >
      <span>
        <span
          className="token plain"
          style={
            {
              "display": "inline-block",
            }
          }
        >
          

        </span>
      </span>
    </div>
  </pre>
</div>
`;

exports[`CODEBLOCKS COMPONENT renders line numbers 1`] = `
<div
  className="my-5 relative group"
>
  <pre
    className="prism-code language-typescript code relative rounded-lg"
    style={
      {
        "backgroundColor": "#011627",
        "color": "#d6deeb",
      }
    }
  >
    <div
      aria-label="Copy to clipboard"
      className="absolute top-3 right-3 group-hover:opacity-100 opacity-0 transform-gpu transition duration-200 group-hover:block"
      title="Copy to clipboard"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-copy fa-fw fa-xl cursor-pointer p-2 bg-slate-500 hover:bg-slate-600 border-2 border-slate-500 hover:border-slate-800 duration-200 rounded-lg active:bg-slate-50"
        data-icon="copy"
        data-prefix="fad"
        focusable="false"
        onClick={[Function]}
        role="img"
        style={{}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          className="fa-duotone-group"
          style={{}}
        >
          <path
            className="fa-secondary"
            d="M192 128H48c-26.5 0-48 21.5-48 48V464c0 26.5 21.5 48 48 48H272c26.5 0 48-21.5 48-48V416H256v32H64V192H192V128z"
            fill="currentColor"
            style={{}}
          />
          <path
            className="fa-primary"
            d="M224 48c0-26.5 21.5-48 48-48H396.1c12.7 0 24.9 5.1 33.9 14.1l67.9 67.9c9 9 14.1 21.2 14.1 33.9V336c0 26.5-21.5 48-48 48H272c-26.5 0-48-21.5-48-48V48z"
            fill="currentColor"
            style={{}}
          />
        </g>
      </svg>
    </div>
    <div
      className="token-line"
      style={
        {
          "backgroundColor": undefined,
          "color": "#d6deeb",
        }
      }
    >
      <span
        className="select-none opacity-50 pr-4 w-11 inline-block text-right"
      >
        1
      </span>
      <span>
        <span
          className="token plain"
          style={
            {
              "display": "inline-block",
            }
          }
        >
          

        </span>
      </span>
    </div>
  </pre>
</div>
`;
